config_system/lex.py

```diff
@@ -16,142 +16,156 @@
+    "DUMMY",
+    "EOL",
+    "EQUAL", "UNEQUAL", "LESS", "LESS_EQUAL", "GREATER", "GREATER_EQUAL",
+    "HELP", "HELPTEXT",
+    "HEX",
+    "IF", "ON",
+    "INT",
+    "LBRACKET", "RBRACKET",
+    "MENU", "ENDMENU", "MAINMENU",
+    "MENUCONFIG",
+    "NUMBER",
+    "PROMPT",
+    "QUOTED_STRING",
+    "SELECT",
+    "SOURCE",
+    "STRING",
+    "TRISTATE",
+    "VISIBLE",
+    "WORD",
+)
 
 states = (
-        ('PARAM', 'exclusive'),
-        ('HELP', 'exclusive'),
-        )
+    ("PARAM", "exclusive"),
+    ("HELP", "exclusive"),
+)
 
 commands = (
-        'bool',
-        'choice',
-        'config',
-        'default',
-        'depends',
-        'endchoice',
-        'endmenu',
-        'int',
-        'menu',
-        'mainmenu',
-        'menuconfig',
-        'prompt',
-        'select',
-        'source',
-        'string',
-        'tristate',
-        'visible',
-        )
-
-params = ("if","on")
+    "bool",
+    "choice",
+    "config",
+    "default",
+    "depends",
+    "endchoice",
+    "endmenu",
+    "int",
+    "menu",
+    "mainmenu",
+    "menuconfig",
+    "prompt",
+    "select",
+    "source",
+    "string",
+    "tristate",
+    "visible",
+)
+
+params = ("if", "on")
 
 help_indent = 0
 
+
 def t_newline(t):
-    r'\n+'
+    r"\n+"
     t.lexer.lineno += len(t.value)
 
+
 def t_ANY_comment(t):
-    r'[ \t]*\#.*'
+    r"[ \t]*\#.*"
     pass
 
+
 def t_blank(t):
-    r'[ \t]+'
+    r"[ \t]+"
+
 
 def t_commandhelp(t):
-    r'help[ \t]*\n'
-    t.lexer.begin('HELP')
+    r"help[ \t]*\n"
+    t.lexer.begin("HELP")
     t.type = "HELP"
     global help_indent
     help_indent = 0
 
     t.lexer.lineno += 1
     return t
 
+
 def t_command(t):
-    r'[A-Za-z0-9_-]+'
-    t.lexer.begin('PARAM')
+    r"[A-Za-z0-9_-]+"
+    t.lexer.begin("PARAM")
     if t.value in commands:
         t.type = t.value.upper()
     else:
         report_error("Unknown identifier %s" % t.value, t)
     return t
 
-t_PARAM_ANDAND = r'&&'
-t_PARAM_OROR = r'\|\|'
-t_PARAM_NOT = r'!'
-t_PARAM_LBRACKET = r'\('
-t_PARAM_RBRACKET = r'\)'
-t_PARAM_EQUAL = r'='
-t_PARAM_UNEQUAL = r'!='
-t_PARAM_LESS = r'<'
-t_PARAM_LESS_EQUAL = r'<='
-t_PARAM_GREATER = r'>'
-t_PARAM_GREATER_EQUAL = r'>='
+
+t_PARAM_ANDAND = r"&&"
+t_PARAM_OROR = r"\|\|"
+t_PARAM_NOT = r"!"
+t_PARAM_LBRACKET = r"\("
+t_PARAM_RBRACKET = r"\)"
+t_PARAM_EQUAL = r"="
+t_PARAM_UNEQUAL = r"!="
+t_PARAM_LESS = r"<"
+t_PARAM_LESS_EQUAL = r"<="
+t_PARAM_GREATER = r">"
+t_PARAM_GREATER_EQUAL = r">="
+
 
 def t_PARAM_blank(t):
-    r'[ \t]+'
+    r"[ \t]+"
     pass
 
+
 def t_PARAM_word(t):
-    r'[A-Za-z][A-Za-z0-9_-]*'
+    r"[A-Za-z][A-Za-z0-9_-]*"
     if t.value in params:
         t.type = t.value.upper()
     else:
-        t.type = 'WORD'
+        t.type = "WORD"
     return t
 
+
 def t_PARAM_number(t):
-    r'[0-9]+'
-    t.type = 'NUMBER'
+    r"[0-9]+"
+    t.type = "NUMBER"
     t.value = int(t.value)
     return t
 
+
 def t_PARAM_string(t):
-    r'"[^"]*"'
+    r'"[^\"]*"'
```

Comment:
Did you intend to add the `\` here?

Reply:
Well actually not. I'll remove this escape character